src/python/pants/util/xml_parser.py

```diff
@@ -30,7 +30,7 @@ def from_file(cls, xml_path):
     try:
       parsed_xml = cls._parse(xml_path)
     except OSError as e:
-      raise XmlParser.XmlError("Problem reading xml file at {}: {}".format(xml_path, e))
+      raise XmlParser.XmlError(f"Problem reading xml file at {xml_path}: {e}")
```

Comment:
In general, great to use `{e!r}` when printing exceptions because it will ensure that the exception class is in the message. We don't do a great job at doing this, but some Pycon talk on exceptions mentioned it's a best practice.